examples/ml-pipelines-node-classification.ipynb

```diff
@@ -0,0 +1,1415 @@
+       "2        47.0  \n",
+       "3        70.0  \n",
+       "4        61.0  \n",
+       "5        18.0  \n",
+       "6        49.0  \n",
+       "7        47.0  \n",
+       "8        28.0  \n",
+       "9        47.0  "
+      ]
+     },
+     "execution_count": 24,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# List of class labels\n",
+    "classes = stats[\"modelInfo\"][\"classes\"]\n",
+    "print(\"Class labels:\", classes)\n",
+    "\n",
+    "# Calculate the confidence percentage for the predicted class\n",
+    "predicted[\"confidence\"] = predicted.apply(\n",
+    "    lambda row: np.floor(row[\"predictedProbabilities\"][classes.index(row[\"predictedClass\"])] * 100), axis=1\n",
+    ")\n",
+    "\n",
+    "predicted"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "rL7Mi1AIc7OJ"
+   },
+   "source": [
+    "## Adding a data preprocessing step\n",
+    "\n",
+    "The quality of the model can potentially be increased by adding more features or by using different features altogether. One way is to use algorithms such as FastRP that create embeddings based on both node properties and graph features, which can be added via the `addNodeProperty` pipeline method. Such properties are \"transient\", in that they are automatically created and removed by the pipeline itself.\n",
+    "\n",
+    "In this example we also use the `contextNodeLabels` parameter to explicitly set the types of nodes we calculate the embeddings for, and we include both the classified and the unclassified nodes. This is useful because the more nodes are used, the better the generated embeddings are. Although it may seem counterintuitive, unclassified nodes do not need to be completely unobserved during training (so, for instance, information on their neighbours can be retained). More information can be found in graph ML publications such as the [Graph Representation Learning Book](https://www.cs.mcgill.ca/~wlh/grl_book/)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "background_save": true
+    },
+    "id": "F_z0gLG9QCAn"
+   },
+   "outputs": [],
+   "source": [
+    "node_pipeline_fastrp, _ = gds.beta.pipeline.nodeClassification.create(\"cora-pipeline-fastrp\")\n",
+    "\n",
+    "# Add a step in the pipeline that mutates the graph\n",
+    "node_pipeline_fastrp.addNodeProperty(\n",
+    "    \"fastRP\",\n",
+    "    mutateProperty=\"embedding\",\n",
+    "    embeddingDimension=512,\n",
+    "    propertyRatio=1.0,\n",
+    "    randomSeed=42,\n",
+    "    featureProperties=[\"features\"],\n",
+    "    contextNodeLabels=[\"Paper\", \"UnclassifiedPaper\"],\n",
+    ")\n",
+    "\n",
+    "# With the node embeddings available as features, we no longer use the original raw `features`.\n",
+    "node_pipeline_fastrp.selectFeatures([\"embedding\"])\n",
+    "\n",
+    "# Configure the pipeline as before\n",
```

Comment:
We _could_ choose to not create a new pipeline, but just re-configure the one we started with. But either works just as well.